Source/WebKit/UIProcess/API/wpe/PageClientImpl.cpp

```diff
@@ -233,6 +233,27 @@ void PageClientImpl::doneWithTouchEvent(const NativeWebTouchEvent& touchEvent, b
             event->modifiers &= ~wpe_input_pointer_modifier_button1;
             page.handleMouseEvent(NativeWebMouseEvent(event, page.deviceScaleFactor()));
         },
+        [&](TouchGestureController::ContextMenuEvent& contextMenuEvent)
+        {
+            auto* event = &contextMenuEvent.event;
+
+            // Mouse motion towards the point of the context click.
+            event->type = wpe_input_pointer_event_type_motion;
+            page.handleMouseEvent(NativeWebMouseEvent(event, page.deviceScaleFactor()));
+
+            event->type = wpe_input_pointer_event_type_button;
+            event->button = 2;
+
+            // Mouse down on the point of the context click.
+            event->state = 1;
+            event->modifiers |= wpe_input_pointer_modifier_button2;
+            page.handleMouseEvent(NativeWebMouseEvent(event, page.deviceScaleFactor()));
+
+            // Mouse up on the same location.
+            event->state = 0;
+            event->modifiers &= ~wpe_input_pointer_modifier_button2;
+            page.handleMouseEvent(NativeWebMouseEvent(event, page.deviceScaleFactor()));
```

Comment:
A possible compromise could be to drop the context click here completely. This way we do not create a click for long tap gestures, which should be alright for the time being. This would at least solve https://bugs.webkit.org/show_bug.cgi?id=251925 for long taps, which is my main motivation.

Reply:
Your choice.

I don't know whether this change is good or not, but it seems reasonable you've had no negative feedback for five months so it's time to land it.

Reply:
Okay, in this case I'll update this PR to drop the context click as suggested above. Thanks!

Reply:
Done.